batch_job/src/work_queue_factory.c

```diff
@@ -1036,14 +1036,13 @@ int main(int argc, char *argv[])
 	}
 	else {
 		if((argc - optind) != 2) {
+			fprintf(stderr,"work_queue_factory: You must either give a project name with the -M option or master-name option with a configuration file, or give the master's host and port.\n");
```

Comment:
I rather have this with an else all by itself. Then it is easier to follow the logic.